pipelines/jenkins/JenkinsAnchoreCICD.txt

```diff
@@ -0,0 +1,98 @@
+                        } catch (Exception ex) {
+                                println("ERROR: Error encountered calling the Anchore endpoint.")
+                                anchoreStatus = "unavailable"
+                        }
+                    }
+                }
+            }
+    
+            stage('Submit image for scanning') {
+                steps {
+                    sh 'echo Running anchorecli'
+                    script {
+                        
+                        // Here we are guarding the pipeline...if anchore is not available, we skip this and log a warning.
+                        if (anchoreStatus == "available") {
+                            println("Anchore is availabile...proceeding.")
+                            
+                            // Here we use credentials configured in the Jenkins credentials store to authenticate to the anchore instance.
+                            withCredentials([usernamePassword(credentialsId: 'Anchore-Pass', passwordVariable: 'pass', usernameVariable: 'user')]) {
+                                
+                                // Submit the image to Anchore for scanning
+                                println("Submitting image for analysis...")
+                                env.digest = sh (script: 'docker run --name anchorecli --rm anchore/anchore-cli-dev anchore-cli --u $user --p $pass --url $ANCHORE_URL --json image add $IMAGENAME":"$TAG | jq -r \'.[] .image_detail[] .digest\'',
+                                    returnStdout: true).trim()
+    
+                                // Check on the status of the scan
+                                println("Checking scan status...for " + digest)
+                                env.status = sh(script: 'docker run --name anchorecli --rm anchore/anchore-cli-dev anchore-cli --u $user --p $pass --url $ANCHORE_URL --json image get $IMAGENAME:$TAG | jq -r \'.[] .analysis_status\'', returnStdout: true).trim()
+                                
+                                // If the scan status indicates not analyzed, we initiate a wait using the timeout configured at the beginning of the pipeline.
+                                println("Analysis Status: " + status)
+                                if (status == "not_analyzed") {
+                                    println("Waiting for image to be analyzed...")
+                                    sh(script: 'docker run --name anchorecli --rm anchore/anchore-cli-dev anchore-cli --u $user --p $pass --url $ANCHORE_URL --json image wait --timeout $WAIT_TIMEOUT $IMAGENAME:$TAG', returnStdout: true)
+                                }
+                            }
+                            
+                        } else {
+                            println("WARNING:: Anchore is not available, skipping scan submission")
+                        }
+                    }
+                }
+            }
+            
+            stage("Verify") {
```

Comment:
Is this where things like `anchore-cli evaluate check` would go?

Reply:
Yep…could be that or whatever other steps that org would want. Maybe I could just change the stage name OR inject a `check` command for completeness.

Reply:
Yeah I feel like a common question is "how do I fail the pipeline on a policy failure" so might be a good idea